src/amazonqFeatureDev/client/featureDev.ts

```diff
@@ -152,11 +154,14 @@ export class FeatureDevClient {
                 assistantResponse.push(responseItem.assistantResponseEvent!.content)
             }
             return assistantResponse.join(' ')
-        } catch (e) {
-            getLogger().error(
-                `${featureName}: failed to execute planning: ${(e as Error).message} RequestId: ${(e as any).requestId}`
+        } catch (e: any) {
+            getLogger().error(`${featureName}: failed to execute planning: ${e.message} RequestId: ${e.requestId}`)
+            throw new ApiError(
+                e.message,
+                'GeneratePlan',
+                e.name,
+                e.$metadata?.httpStatusCode ?? streamResponseErrors[e.name]
```

Comment:
I'm kind of confused with what streamResponseErrors does. Shouldn't the incoming errors already have the correct httpStatusCode set? I would think things like AccessDeniedException would already be throwing a 403 and thus it's not really needed in that mapping

Reply:
This is just a map to prepare if in the future RTS would handle midstreaming errors differently. Basically incoming errors thrown before streaming, i.e. validateTaskAssist, will throw an error we can get the status code from. This is no the case for midstream errors which are just translated to an InternalServerError with no extra metadata.

Reply:
Not really, mid-stream errors are only thrown with a name. I improved the error parsing in the later commits alluding to a comment about not using e: any, that will make more clear what the shape of the error is.